File: CarApp/app/src/main/java/accelero/Accelerometer.java
package accelero;

import  android.app.*;
import android.hardware.*;
import android.widget.*;
import android.os.*;
import android.content.*;

import org.w3c.dom.Text;

public class Accelerometer extends Activity implements SensorEventListener {
    StringBuilder builder = new StringBuilder();

    float [] history = new float[2];
    String [] direction = {"NONE","NONE"};

    @Override
    public Accelerometer() {

        SensorManager manager = (SensorManager) getSystemService(Context.SENSOR_SERVICE);
        Sensor accelerometer = manager.getSensorList(Sensor.TYPE_ACCELEROMETER).get(0);
        manager.registerListener(this, accelerometer, SensorManager.SENSOR_DELAY_GAME);
    }

    @Override
    public void onSensorChanged(SensorEvent event, TextView text) {

        float xChange = history[0] - event.values[0];
        float yChange = history[1] - event.values[1];

        history[0] = event.values[0];
        history[1] = event.values[1];

        if (xChange > 2){
            direction[0] = "LEFT";
        }
        else if (xChange < -2){
            direction[0] = "RIGHT";
        }

        if (yChange > 2){
            direction[1] = "DOWN";
        }
        else if (yChange < -2){
            direction[1] = "UP";
        }

        builder.setLength(0);
        builder.append("x: ");
        builder.append(direction[0]);
        builder.append(" y: ");
        builder.append(direction[1]);

        text.setText(builder.toString());
    }

    @Override
    public void onAccuracyChanged(Sensor sensor, int accuracy) {
        // nothing to do here
    }
}

File: RpcApp/app/src/main/java/com/car_rpc/app/Accelerometer.kt
package com.car_rpc.app

import android.content.Context
import android.content.pm.ActivityInfo
import android.hardware.Sensor
import android.hardware.SensorEvent
import android.hardware.SensorEventListener
import android.hardware.SensorManager
import android.os.Bundle
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import android.view.View

class Accelerometer : AppCompatActivity(), SensorEventListener {
    private var manager: SensorManager? = null
    private lateinit var accelerometer: Sensor
    internal lateinit var coordView: TextView
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_PORTRAIT
        coordView = findViewById<View>(R.id.coordView) as TextView
        manager = getSystemService(Context.SENSOR_SERVICE) as SensorManager
        accelerometer = manager!!.getDefaultSensor(Sensor.TYPE_ACCELEROMETER)
        manager!!.registerListener(this@Accelerometer, accelerometer, SensorManager.SENSOR_DELAY_NORMAL)

    }

    override fun onSensorChanged(event: SensorEvent) {
        coordView.text = " ACCELEROMETER\n" + " X: " + event.values[0] + "\n Y: " + event.values[1] + "\n Z: " + event.values[2]
    }

    override fun onAccuracyChanged(sensor: Sensor, i: Int) {

    }
}